AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomGPU_BasicLevelSetup.py

```diff
@@ -0,0 +1,288 @@
+    """
+    Summary:
+    Sets up a level to match the AtomBasicLevelSetup.ppm golden image then takes a screenshot to verify the setup.
+
+    Test setup:
+    - Wait for Editor idle loop.
+    - Open the "Base" level.
+
+    Expected Behavior:
+    The scene can be setup for a basic level.
+    The test screenshot matches the appearance of the AtomBasicLevelSetup.ppm golden image.
+
+    Test Steps:
+    1. Wait for the level to load and update the viewport size.
+    2. Create Default Level Entity.
+    3. Create Grid Entity as a child entity of the Default Level Entity.
+    4. Add Grid component to Grid Entity and set Secondary Grid Spacing.
+    5. Create Global Skylight (IBL) Entity as a child entity of the Default Level Entity.
+    6. Add HDRi Skybox component to the Global Skylight (IBL) Entity.
+    7. Add Global Skylight (IBL) component to the Global Skylight (IBL) Entity.
+    8. Set the Cubemap Texture property of the HDRi Skybox component.
+    9. Set the Diffuse Image property of the Global Skylight (IBL) component.
+    10. Set the Specular Image property of the Global Skylight (IBL) component.
+    11. Create a Ground Plane Entity with a Material component that is a child entity of the Default Level Entity.
+    12. Set the Material Asset property of the Material component for the Ground Plane Entity.
+    13. Add the Mesh component to the Ground Plane Entity and set the Mesh component Mesh Asset property.
+    14. Create a Directional Light Entity as a child entity of the Default Level Entity.
+    15. Add Directional Light component to Directional Light Entity and set entity rotation.
+    16. Create a Sphere Entity as a child entity of the Default Level Entity then add a Material component.
+    17. Set the Material Asset property of the Material component for the Sphere Entity.
+    18. Add Mesh component to Sphere Entity and set the Mesh Asset property for the Mesh component.
+    19. Create a Camera Entity as a child entity of the Default Level Entity then add a Camera component.
+    20. Set the Camera Entity rotation value and set the Camera component Field of View value.
+    21. Enter game mode.
+    22. Take screenshot.
+    23. Exit game mode.
+    24. Look for errors.
+
+    :return: None
+    """
+
+    import os
+    from math import isclose
+
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+    import azlmbr.paths
+
+    from editor_python_test_tools.asset_utils import Asset
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
+
+    from Atom.atom_utils.screenshot_utils import ScreenshotHelper
+
+    MATERIAL_COMPONENT_NAME = "Material"
+    MESH_COMPONENT_NAME = "Mesh"
+    SCREENSHOT_NAME = "AtomBasicLevelSetup"
+    SCREEN_WIDTH = 1280
+    SCREEN_HEIGHT = 720
+    DEGREE_RADIAN_FACTOR = 0.0174533
+
+    def initial_viewport_setup(screen_width, screen_height):
```

Comment:
Is the idea here that you are validating the viewport size change with isclose check? But most tests will not need to use the wait_for_condition, they can just call set_viewport_size followed by update_viewport?

If you'd expect most Atom tests (or at least those with screenshot comparison) to need to wait for the screen size to change, then consider moving this to be part of the ScreenshotHelper or atom_utils

Reply:
removing from the function